packages/block-library/src/image/icon.js

```diff
@@ -3,8 +3,8 @@
  */
 import { Path, SVG } from '@wordpress/components';
 
-function svg( props ) {
+export function SvgIcon( props ) {
 	return <SVG viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" { ...props }><Path d="M0,0h24v24H0V0z" fill="none" /><Path d="m19 5v14h-14v-14h14m0-2h-14c-1.1 0-2 0.9-2 2v14c0 1.1 0.9 2 2 2h14c1.1 0 2-0.9 2-2v-14c0-1.1-0.9-2-2-2z" /><Path d="m14.14 11.86l-3 3.87-2.14-2.59-3 3.86h12l-3.86-5.14z" /></SVG>;
 }
 
-export default svg;
+export default SvgIcon( {} );
```

Comment:
I know these icon.js files are supposed to be normalized across all block types. Both the previous and the new notations seem wrong to me (compared to the other files)

cc @gziolo

Reply:
I think the idea is to revert this in #15778, and this is just a temporary fix for this week's mobile release. @hypest should we create a 1.6 branch and merge this there instead of master? We can then revert the icon changes as part of #15778

Reply:
In other places, we do something as follows:

```js
return (
    <SVG viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><Path d="M0,0h24v24H0V0z" fill="none" /><Path d="m19 5v14h-14v-14h14m0-2h-14c-1.1 0-2 0.9-2 2v14c0 1.1 0.9 2 2 2h14c1.1 0 2-0.9 2-2v-14c0-1.1-0.9-2-2-2z" /><Path d="m14.14 11.86l-3 3.87-2.14-2.59-3 3.86h12l-3.86-5.14z" /></SVG>;
);
```

it's an element rather than a component. If that's an issue we can think about ways to update all block icons.

Reply:
Technically speaking, you can use `cloneElement( icon, extraProps  )` to inject more props as well.

Reply:
Changed the base branch to `rnmobile/release-v1.6.0` and noted the required changes in the upcoming PR: https://github.com/WordPress/gutenberg/pull/15778#issuecomment-496828916

Reply:
> Technically speaking, you can use cloneElement( icon, extraProps ) to inject more props as well.

I used that API in [here](https://github.com/WordPress/gutenberg/pull/15769/files#diff-8033adfe49351c67798c048d64dda906R8) and to be honest, it felt kinda awkward, a bit hacky and probably not a well maintainable solution. AFAICT, the Gutenberg code leans more on HOCs to do augmentations on components and we probably should try something similar here? I.e. perhaps the assumptions of the Icon code [here](https://github.com/WordPress/gutenberg/blob/3317476e6d3d6cf3bb4c9634f377cede661a3aa0/packages/components/src/icon/index.js#L31) are too strict and inflexible?

Reply:
I guess we could replace that check with extending the `SVG` component to translate the `size` prop into `width`/`height`

Reply:
I'm going to review approve and merge this PR (to the release branch, not `master`) if it's working OK. We'll need to work on a separate PR to make it mergable to `master`.